gnomad/utils/constraint.py

```diff
@@ -372,6 +372,42 @@ def _get_criteria(i: hl.expr.Int32Expression) -> hl.expr.Int32Expression:
     return hl.agg.array_sum(hl.map(_get_criteria, sorted_indices))
 
 
+def explode_downsamplings(
+    ht: hl.Table,
+    downsampling_meta: Dict[str, List[str]],
+    metrics: List[str] = ["syn", "lof", "mis"],
+) -> hl.Table:
+    """
+    Explode downsampling counts for each genetic ancestry group and metric.
+
+    :param ht: Input Table. Should contain struct of downsampling information for genetic ancestry groups under each metric name. Example: 'lof': struct {gen_anc_exp: struct {global: array<float64>}.
+    :param metrics: List of metrics to explode. Default is '['syn', 'lof', 'mis']'.
+    :param downsampling_meta: Dictionary containing downsampling metadata. Keys are the genetic ancestry group names and values are the list of downsamplings for that genetic ancestry group. Example: {'global': ['5000', '10000'], 'afr': ['5000', '10000']}.
+    :return: Table with downsampling counts exploded so that observed and expected metric counts for each pair of genetic ancestry groups and downsamplings is a separate row.
+    """
+    ht = ht.select(
+        data=[
+            hl.struct(
+                gen_anc=pop,
+                downsampling=downsampling,
+                **{
+                    f"{metric}.exp": ht[metric]["gen_anc_exp"][pop][i]
+                    for metric in metrics
+                },
+                **{
+                    f"{metric}.obs": ht[metric]["gen_anc_obs"][pop][i]
+                    for metric in metrics
+                },
```

Comment:
Since these are not different anymore, can simplify with this:

```suggestion
                **{
                    f"{metric}.{oe}": ht[metric][f"gen_anc_{oe}"][pop][i]
                    for metric in metrics
                    for oe in ["exp", "obs"]
                },
```

However, in gnomad_lof it looks like it was:
```
hl.cond(hl.is_missing(ht[f'obs_lof_{pop}'][i]) & hl.is_defined(ht.obs_lof), 0, ht[f'obs_lof_{pop}'][i])
```
Do we need to check that the non pop obs is defined?

Reply:
with the way the pipeline is written i don't think we ever get missing values here